### PR TITLE
CS

### PR DESCRIPTION
**Pull request by leofeyer**
Since there are quite a few CS fixes, I decided to make this a PR.

### CODE REVIEW

File: core-bundle/src/Doctrine/Backup/BackupManager.php [hunk 1]
```diff
@@ -190,12 +190,10 @@ private function finishWriting($fileHandle, $deflateContext): void
         if ($deflateContext) {
             fwrite($fileHandle, deflate_add($deflateContext, '', ZLIB_FINISH));
         }
+
         fclose($fileHandle);
     }
 
-    /**
-     * @throws BackupManagerException
-     */
```
Comment (Toflar) [inline core-bundle/src/Doctrine/Backup/BackupManager.php:198]: Why is this removed? My IDE wants that to be present so it knows exceptions are handled properly :)

File: core-bundle/src/Doctrine/Backup/BackupManager.php [hunk 2]
```diff
@@ -247,13 +245,11 @@ private function tidyDirectory(): void
             if ($i >= $this->keepMax) {
                 (new Filesystem())->remove($backup->getFilepath());
             }
+
             ++$i;
         }
     }
 
-    /**
-     * @throws BackupManagerException
```
Comment (Toflar) [inline core-bundle/src/Doctrine/Backup/BackupManager.php:255]: Same here

File: core-bundle/src/Doctrine/Backup/Config/AbstractConfig.php [hunk 3]
```diff
@@ -43,32 +43,23 @@ public function isGzCompressionEnabled(): bool
         return $this->gzCompression;
     }
 
-    /**
-     * @return static
-     */
-    public function withGzCompression(bool $enable)
```
Comment (Toflar) [inline core-bundle/src/Doctrine/Backup/Config/AbstractConfig.php]: How were you able to get this working with phpStan? I explicitly did this so it works.

File: core-bundle/src/Doctrine/Backup/BackupManager.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Doctrine/Backup/BackupManager.php:198]: We are not using `@throws` annotations, because they are mostly not maintained correctly and therefore outdated.  You might want to adjust your PHPStorm configuration:

File: core-bundle/src/Doctrine/Backup/Config/AbstractConfig.php (hunk 3 above)
Comment (leofeyer) [reply core-bundle/src/Doctrine/Backup/Config/AbstractConfig.php]: I have already reverted these changes. Guess we have to wait for PHP 8, so we can use `: static`.

File: core-bundle/src/Doctrine/Backup/BackupManager.php (hunk 1 above)
Comment (m-vo) [reply core-bundle/src/Doctrine/Backup/BackupManager.php:198]: We should maybe discuss this again at some point. Annotating exceptions that are thrown as part of the business logic is IMHO good practice. (This should *only* cover custom exceptions).

Comment (leofeyer) [reply core-bundle/src/Doctrine/Backup/BackupManager.php:198]: I disagree. PHP is moving away from phpDoc comments for a reason and so should we.

phpDoc comments are very error-prone, because they are not enforceable, and they tend to be forgotten when things are refactored. We had hundreds of wrong phpDoc comments in the code base and I am sure we still have some.

A modern static PHP analyzer will know which exceptions are thrown without a `@exception` annotation.

Comment (m-vo) [reply core-bundle/src/Doctrine/Backup/BackupManager.php:198]: I get your point about maintainability but it's IMHO also about API design.

Here is an example:
```php
/**
 * @throws CalculationSkippedException if the calculation was skipped
 */
function calculateSomething(Config $config): int {
    if(!$this->validateConfig($config)) {
        throw new InvalidConfigurationException('bad config');
    }

    // …
    throw new CalculationSkippedException();
}
```

How would you use this API? With the annotation it would be clear that one of the exceptions *must always* be handled because it does not indicate an error:

```php
try { 
    $result = $calculator->calculateSomething($myConfig);
}
catch (CalculationSkippedException) {
    $result = 0;
}
```

Comment (Toflar) [reply core-bundle/src/Doctrine/Backup/BackupManager.php:198]: Guys, you are stopping a PR from being merged. Please discuss this separately because it's not related. We don't have it anywhere, so we don't need it here either.

Comment (m-vo) [reply core-bundle/src/Doctrine/Backup/BackupManager.php:198]: 🤷‍♂️

Comment (leofeyer) [reply core-bundle/src/Doctrine/Backup/BackupManager.php:198]: PHPStorm does not need a phpDoc to handle your example just fine:
